File: OffByOne/src/OffByOne.Qls/Ast/Style/Statements/StyleSheet.cs
namespace OffByOne.Qls.Ast.Style.Statements
{
    using System;
    using System.Collections.Generic;

    using MoreDotNet.Wrappers;

    using OffByOne.Qls.Ast.Style.Statements.Base;
    using OffByOne.Qls.Common.Visitors.Contracts;

    public class StyleSheet : Statement
    {
        public StyleSheet(
            string id,
            IEnumerable<Page> pages = null)
        {
            if (id.IsNullOrWhiteSpace())
            {
                throw new ArgumentException("A non-null, non-empty identifier must be given.");
            }

            this.Id = id;
            this.Pages = pages;
        }

        public string Id { get; }

        public IEnumerable<Page> Pages { get; private set; }

        public override TResult Accept<TResult, TContext>(
            IStatementVisitor<TResult, TContext> visitor,
            TContext environment)
        {
            return visitor.Visit(this, environment);
        }
    }
}

File: Loading/qls/src/qls/ast/Stylesheet.java
package qls.ast;

import ql.ast.Node;

import java.util.Iterator;
import java.util.List;

public class Stylesheet extends Node implements Iterable<Page> {

    private final List<Page> pages;

	public Stylesheet(List<Page> pages, int line) {
		super(line);
		this.pages = pages;
	}
	
	public void accept(StylesheetVisitor v) {
		v.visit(this);
	}

	@Override
	public Iterator<Page> iterator() {
		return pages.iterator();
	}

	

}
